### ISSUE TITLE
writeln! and println! (and builtin macros’?) diagnostics are terrible

### ISSUE THREAD

**Issue by nagisa** [rust-lang/rust#23459]
```
<std macros>:4:19: 4:46 error: invalid reference to argument `1` (there is 1 argument)
<std macros>:4 write ! ( $ dst , concat ! ( $ fmt , "\n" ) , $ ( $ arg ) * ) ) ;
                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~

<std macros>:3:11: 3:38 error: invalid reference to argument `2` (there are 2 arguments)
<std macros>:3 print ! ( concat ! ( $ fmt , "\n" ) , $ ( $ arg ) * ) ) ;
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I’d expect it to at least emit an `in expansion of writeln!` and `in expansion of println!` hints which it doesn’t currently.

**Comment by jakerr** [comment]
It also doesn't even say where the invocation is. With `cargo build --verbose` I can see which file was being compiled when a macro invocation broke but I have no real idea which line it was. When there are many similar invocations it can be quite painful to track down.

**Comment by pornel** [comment]
:+1: 

```
println!("{} {}", 1);
```

> \<std macros>:3 print ! ( concat ! ( $ fmt , "\n" ) , $ ( $ arg ) * ) ) ;

**Comment by tamird** [comment]
This should have been closed by #24003 I believe
